src/sage/combinat/lp_algebra_seed.py

```diff
@@ -0,0 +1,1207 @@
+
+        - ``algorithm`` -- String (default: ``'BFS'``); the search algorithm to
+          find new seeds. Currently supported options::
+
+          * 'BFS' - breadth-first search
+          * 'DFS' - depth-first search
+
+        EXAMPLES:
+
+        We iterate over the mutation class for a rank two seed::
+
+            sage: var('x1,x2')
+            (x1, x2)
+            sage: S = LPASeed({x1: 1 + x2, x2: 1 + x1})
+            sage: t = S.mutation_class_iter()
+            sage: for seed in t: print(seed.cluster())
+            (x1, x2)
+            ((x2 + 1)/x1, x2)
+            (x1, (x1 + 1)/x2)
+            ((x2 + 1)/x1, (x1 + x2 + 1)/(x1*x2))
+            ((x1 + x2 + 1)/(x1*x2), (x1 + 1)/x2)
+
+        Non finite-type works if we specify a fixed depth, but seeds can get big
+        rather quickly::
+
+            sage: var('x1,x2')
+            (x1, x2)
+            sage: S = LPASeed({x1: 1 + x2 + x2^2, x2: 1 + x1 + x1^2})
+            sage: t = S.mutation_class_iter(depth=15,algorithm='DFS')
+            sage: for seed in t: print(seed.cluster()[0].denominator()) # long time
+            1
+            x1
+            1
+            x1*x2^2
+            x1*x2^2
+            x1^3*x2^4
+            x1^3*x2^4
+            x1^5*x2^6
+            x1^5*x2^6
+            x1^7*x2^8
+            x1^7*x2^8
+            x1^9*x2^10
+            x1^9*x2^10
+            x1^11*x2^12
+            x1^11*x2^12
+            x1^13*x2^14
+
+        We can print computational statistics when computing examples::
+
+            sage: var('x1,x2,x3,a0,a2,a3,b0,b1,b3,c0,c1,c2')
+            (x1, x2, x3, a0, a2, a3, b0, b1, b3, c0, c1, c2)
+            sage: S = LPASeed({x1: a0 + a2*x2 + a3*x3, x2: b0 + b1*x1 + b3*x3, x3: c0 + c1*x1 + c2*x2})
+            sage: t = S.mutation_class_iter(verbose=True)
+            sage: for seed in t: None
+            depth: 0     found: 1
+            depth: 1     found: 4
+            depth: 2     found: 10
+            depth: 3     found: 16
+        """
+
+        # initialise
+
+        n = self._rank
+        seeds_found = [self]
+        current_depth = 0
+        seeds_to_check = [self]
+
+        # If we are showing depth, show some statistics
+        if verbose:
+            dc = str(current_depth)
+            dc += ' ' * (5 - len(dc))
+            nr = str(len(seeds_found))
+            nr += ' ' * (10 - len(nr))
+            print("depth: %s found: %s" % (dc, nr))
+
+        if return_paths:
+            yield (self, [])
+        else:
+            yield self
+
+        if algorithm == 'BFS':
+
+            new_seeds_found = True
+
+            while (new_seeds_found and current_depth < depth):
+
+                current_depth += 1
+                new_seeds = []  # reset new seed list
+
+                for seed in seeds_to_check:
+
+                    # we do not need to check the index we last mutated at
+                    if seed._mutation_sequence == []:
```

Comment:
```suggestion
                    if not seed._mutation_sequence:
```
A little faster this way.